templates/alertmanager/alertmanager-statefulset.yaml

```
@@ -21,6 +21,11 @@ spec:
     {{- toYaml .Values.alertmanager.statefulStrategy | nindent 4 }}
   serviceName: {{ template "cortex.fullname" . }}-alertmanager-headless
   {{- if .Values.alertmanager.persistentVolume.enabled }}
+  {{- if and (semverCompare ">=1.23-0" .Capabilities.KubeVersion.Version) (.Values.alertmanager.persistentVolume.enableRetentionPolicy) }}
+  persistentVolumeClaimRetentionPolicy:
+    whenDeleted: {{ .Values.alertmanager.persistentVolume.whenDeleted }}
+    whenScaled: {{ .Values.alertmanager.persistentVolume.whenScaled }}
+  {{- end }}
```

Do we get any benefit from having the bool `enableRetentionPolicy` to enable this? To me it seems simpler to just use the values if they are defined, but I get that this is a bit of a style thing.

It could be like this:
```suggestion
  {{- if (semverCompare ">=1.23-0" .Capabilities.KubeVersion.Version) }}
  {{- with .Values.alertmanager.persistentVolume.retentionPolicy }}
  persistentVolumeClaimRetentionPolicy:
    {{- toYaml . | nindent 4}}
  {{- end }}
  {{- end }}
```
The values.yaml would be like:
```yaml
alertmanager:
  persistentVolume:
    retentionPolicy: {}
    # whenDeleted: Retain
    # whenScaled: Retain
```

---

I'm fine with both.

---

@kd7lxl I like that suggestion, I'll go ahead and make the changes